File: number_of_ones.c
#include <stdio.h>
int main(int argc, const char * argv[]) {
	FILE *file = fopen(argv[1], "r");
	char line[1024];
	while (fgets(line, 1024, file)) {
		int n=atoi(line);
		int sum=0;
		while (n) {
			if (n%2) {
				n--;
				sum++;
			}
			else {n/=2;}
		}
		printf("%d\n",sum);
	}
	return 0;
}

File: number_of_ones.lua
for line in io.lines(arg[1]) do
	n=tonumber(line)
	sum=0
	while n>0 do
		if n%2==0 then
			n=n/2
		else
			n=n-1
			sum=sum+1
		end
	end
	print(sum)
end
